content/oci/readonlyoci.go

```diff
@@ -224,3 +224,29 @@ func listTags(ctx context.Context, tagResolver *resolver.Memory, last string, fn
 
 	return fn(tags)
 }
+
+// stripAnnotationRefName strips the AnnotationRefName from the annotation map
+// of desc.
+func stripAnnotationRefName(desc ocispec.Descriptor) ocispec.Descriptor {
+	if _, ok := desc.Annotations[ocispec.AnnotationRefName]; !ok {
+		// no ops
+		return desc
+	}
+
+	l := len(desc.Annotations)
+	if l == 1 {
+		// the annotation map contains ref name only
+		desc.Annotations = nil
+		return desc
+	}
+
+	// the annotation map contains more than 1 key
+	annotations := make(map[string]string, l-1)
```

Comment:
It would be better to have a meaning variable name other than `l`.

```suggestion
	size := len(desc.Annotations) - 1
	if size == 0 {
		desc.Annotations = nil
		return desc
	}
	annotations := make(map[string]string, size)
```